buildSrc/src/main/groovy/com/uber/okbuck/core/model/android/AndroidTarget.groovy

```diff
@@ -177,11 +177,11 @@ abstract class AndroidTarget extends JavaLibTarget {
 
         Map<String, ClassField> extraBuildConfig = [:]
 
-        baseVariant.buildType.buildConfigFields.collect { String key, ClassField classField ->
+        baseVariant.mergedFlavor.buildConfigFields.collect { String key, ClassField classField ->
```

Comment:
```
android {
  defaultConfig {
    buildConfigField "Integer", "VALUE", "1"
  }

  buildTypes {
    debug {
      buildConfigField "Integer", "VALUE", "2"
    }
  }

  productFlavors {
    full {
      buildConfigField "Integer", "VALUE", "3"
    }
  }
}
```

In the above case, my change will cause the end result to be 2 when it should be 3. Good catch. However, without the flavor, the behavior isn't as expected without this change.